File: ep-2-tooling/code/comparison/go/for_loop.go
package main

import "fmt"

func for_loop_implementation() {
    a := []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
    b := []int{}
    new_val := 0
    for _, n := range a {
        new_val = n * 2
        if new_val % 3 == 0 {
            b = append(b, new_val)
        }
    }

    fmt.Println("For Loop Output:", b)
}

File: ep-2-tooling/code/comparison/python/for_loop.py
# Take a list of positive integers < 10:
#   * Multiply each integer by 3
#   * Filter any values that aren't divisible by 3

def implementation():
    a = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
    b = []

    for n in a:
        new_val = n * 2
        if new_val % 3 == 0:
            b.append(new_val)

    print("For Loop Output:", b) # Expected output: [0, 6, 12, 18]
